### ISSUE TITLE
Showstopper: Missing axis labels

### ISSUE THREAD

**Issue by susburrows** [UV-CDAT/uvcmetrics#68]
Fixing the missing axis labels in many plots (numerical labels for lat/lon/height axes) is a high priority, since the plots can be difficult to interpret without axis labels.

**Comment by doutriaux1** [comment]
@susburrows @painter1 I'm going to take a look at this. Can you please provide an example on how to reproduce this (command line args). Thx.

**Comment by susburrows** [comment]
metadiags.py --package amwg --model /dtemp/burr114/climos/famipc5_ne30_v0.3_ens_latlon --obspath /home/burr114/obs_data --outpath /dtemp/burr114/diags/famipc5_ne30_v0.3_ens_latlon_uvcdat_v2.2_uvcmetrics-devel --hostname acme-dev-2.ornl.gov --dsname famipc5_ne30_v0.3_ens_latlon --colls topten

(obviously the paths may need to be adjusted)

**Comment by doutriaux1** [comment]
thx! That should do it.

**Comment by doutriaux1** [comment]
@painter1 also posted:
```
diags --outputdir ~/tmp/diagout/ --model path=/Users/painter1/metrics_data/cam35_data/,climos=yes --obs path=~/metrics_data/obs_data_5.6/,filter="f_startswith('NCEP')",climos=yes --varopts 850 --package AMWG --set 5 --var T --seasons ANN
```

**Comment by doutriaux1** [comment]
good news on my system i can reproduce with:
```
diags --outputdir /Users/doutriaux1/diagout/ --model path=/Users/doutriaux1/uvcmetrics_test_data/cam_output/,climos=yes --obs path=/Users/doutriaux1/uvcmetrics_test_data/obs/,filter="f_startswith('NCEP')",climos=yes --varopts 850 --package AMWG --set 5 --var T --seasons ANN
```
![set5_global_ann_t-combined](https://cloud.githubusercontent.com/assets/2781425/10441163/be59edca-70fa-11e5-8cf5-0627aee71ffb.png)

**Comment by doutriaux1** [comment]
@susburrows would the following be ok?
![set5_global_ann_t-combined](https://cloud.githubusercontent.com/assets/2781425/10441549/021a73a0-70ff-11e5-93cb-dd88a1efeba5.png)

**Comment by susburrows** [comment]
Glad to hear you could reproduce the issue.  I hope it is easy to resolve.

**Comment by doutriaux1** [comment]
@susburrows it's easy to fix, would the pix above work for you?

**Comment by susburrows** [comment]
Thanks Charles, that's great that it is easy to fix.  Could you also add axis labels (lat/lon)?  Also, I'm not sure if labels are present or not on all the other plots (other than set5), I'd have to check.

**Comment by doutriaux1** [comment]
will do, should I add the xaxis label (longitude) on all 3 plots or only the last one?

**Comment by susburrows** [comment]
Thanks!  Well, the package produces individual plots as well as panel plots.  I think the (lat/lon) labels should appear on each of the individual plots.  On the panel plots, it would be ok if the xaxis label is only on the last one, as you suggest, since they all have the same axis.

**Comment by doutriaux1** [comment]
@susburrows @painter1 would you mind building off the `issue_68_missing_axes_labels` branch and let me know if that looks better and/or if I broke something else. Thx.

**Comment by susburrows** [comment]
Hi @doutriaux1 , thanks for working on this.  It is not entirely working for me right now.
I'm using the most recent version of the issue_68_missing_axes_labels branch
(ca1f6744c2ce9831a6a9241bb2efd2ff39699c9b), and I'm running on cascade, using UV-CDAT 2.2 (with your branch of uvcmetrics).

One issue I am seeing is that on the Set3 plots, the vertical axis has no label (I think it should be the name of the field, FLDS in the example below), and the "lat" label is showing up inside the plot area -- I think it should be outside of (below) the plot area.  I think I am still seeing issues in some of the other plots, also.  I'll have a look and let you know in further comments.
![settier1b_so_ann_flds_isccp_s_extratropics-model-and-obs](https://cloud.githubusercontent.com/assets/6811403/10527118/9abccf04-7343-11e5-86a0-2a6ed00e2d29.png)

**Comment by susburrows** [comment]
So, In the set5 plots (for the tier1b_so collection), on the individual plots, I'm getting axis (lat/lon) labels for the individual plots, but no tick marks.  In the combined plot, there are no labels or tick marks.

I'm not sure if this is specific to the tier1b_so collection, or to plots with regional selection perhaps.  I'll try the topten collection also.

![settier1b_so_ann_swcf_ceres-ebaf_s_extratropics-model-and-obs](https://cloud.githubusercontent.com/assets/6811403/10527287/8832ca5e-7344-11e5-9a1f-d5681175c367.png)

![settier1b_so_ann_swcf_ceres-ebaf_s_extratropics-combined](https://cloud.githubusercontent.com/assets/6811403/10527392/08ab00b6-7345-11e5-8cfe-bbce796908ad.png)

**Comment by susburrows** [comment]
@doutriaux1 , I still get similar results for the global plots in the topten collection, which I called through metadiags.py (which calls diags-new.py).  Are you getting different results when you run the package?

![settopten_ann_lwcf_ceres-ebaf__global-model-and-obs](https://cloud.githubusercontent.com/assets/6811403/10527825/7972a536-7347-11e5-93a0-84cd3838a1eb.png)
![settopten_ann_prect_gpcp_global-combined](https://cloud.githubusercontent.com/assets/6811403/10527835/8aa55466-7347-11e5-9a11-0abc04883e29.png)

**Comment by doutriaux1** [comment]
@susburrows I talked about this with @williams13 in some cases it was originally required that the plots be EXACTLY as AMWG and apparently AMWG did not have ticks. So we have some "DUD" templates with evrything off. I will double check when we actually use these. Because the "DUD" templates are really meant to be such, I did not turned back ticks on these. @painter1 do you know if there's is a flag in the code to use either one? Thanks @susburrows I will try to run the whole suite and look at these and further tweak it.

**Comment by painter1** [comment]
No, I don't know of any flag to switch between tic marks and not, or anything more general such as "exactly like AMWG" or not.

**Comment by susburrows** [comment]
So the question this raises for me is, where did the requirement that it be "exactly like AMWG" (even where AMWG is suboptimal) come from?  And, could it be relaxed at this point?

**Comment by doutriaux1** [comment]
ask @williams13 and @painter1 they were in charge then.

**Comment by williams13** [comment]
@painter1 is the lead of diagnostics. His call.

**Comment by painter1** [comment]
The original "exactly like AMWG" goal was motivated by the feeling that many users are extremely conservative and couldn't accept anything which didn't look familiar.  But even at the beginning, "exactly like AMWG" only applied to superficial things like the presence of tic marks.  On issues of features, capabilities, flexibility, code structure, and correctness our goals have always been to different and far better.

AMWG is no longer a goal or constraint.

**Comment by doutriaux1** [comment]
@painter1 when running @susburrows command line I get:
```
metadiags.py --package amwg --model /Users/doutriaux1/famipc5_ne30_v0.3_00003.cam.h0.1979-01.nc --obspath=/Users/doutriaux1/uvcmetrics_test_data/obs --outpath=/Users/doutriaux1/diags_out --hostname acme-dev-2.ornl.gov --dsname famipc5_ne30_v0.3_ens_latlon --colls topten
could not load tmpl: UVWG Error: texlm not a valid textorientation
/lgm/uvcdat/2015-10-02/Library/Frameworks/Python.framework/Versions/2.7/bin/metadiags.py
usage: metadiags.py [options]
metadiags.py: error: unrecognized arguments: --obspath=/Users/doutriaux1/uvcmetrics_test_data/obs --outpath=/Users/doutriaux1/diags_out
```
Any suggestion?

**Comment by doutriaux1** [comment]
@painter1 canI sit with you on this on Monday? It will be faster, you know where everything is set in diags/diags_new_metadiag and I know how to fix things.

**Comment by painter1** [comment]
I'm available for that, of course.  But Brian, not me, wrote metadiags.  In fact, I have never run metadiags!  Your problem is simply that Brian changed the metadiags syntax.  The new syntax for paths looks like
--model path=<path-to-model>,climos=yes --obs path=<path-to-obs>,climos=yes --outputdir <outputdir>
See the ACME Confluence page on plot labels, where Brian says that he made a change:
https://acme-climate.atlassian.net/wiki/display/WORKFLOW/Missing+axis+labels+on+uvcmetrics+plots?focusedCommentId=36865138#comment-36865138

**Comment by doutriaux1** [comment]
thx @painter1 I'm getting passed this now. Will probably still need to sit with you on monday though.

**Comment by doutriaux1** [comment]
@painter1 not sure this is done, far from it even.

**Comment by painter1** [comment]
I don't know how this got closed, but it was surely a mistake.  Maybe I thought I was typing in another window or something.
